### PR TITLE
Allow admin to query without channel slug

### PR DESCRIPTION
**Pull request by d-wysocki**
I want to merge this change because...

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/checkout/resolvers.py [hunk 1]
```diff
@@ -8,8 +8,10 @@ def resolve_checkout_lines():
     return queryset
 
 
-def resolve_checkouts():
+def resolve_checkouts(channel_slug):
     queryset = models.Checkout.objects.all()
+    if channel_slug:
+        queryset = queryset.filter(channel__slug=str(channel_slug))
```
Comment (fowczarek) [inline saleor/graphql/checkout/resolvers.py:14]: Maybe? 
In this revoler we have only `channel_slug` as `str`. `str()` is required only if we have options to pass `SimpleLesyObject` to `filter`.
```suggestion
        queryset = queryset.filter(channel__slug=channel_slug)
```

File: saleor/graphql/discount/resolvers.py [hunk 2]
```diff
@@ -9,10 +9,14 @@
 def resolve_vouchers(info, query, channel_slug, **_kwargs) -> ChannelQsContext:
     qs = models.Voucher.objects.all()
     qs = filter_by_query_param(qs, query, VOUCHER_SEARCH_FIELDS)
+    if channel_slug:
+        qs = qs.filter(channel_listing__channel__slug=str(channel_slug))
```
Comment (fowczarek) [inline saleor/graphql/discount/resolvers.py:13]:
```suggestion
        qs = qs.filter(channel_listing__channel__slug=channel_slug)
```

File: saleor/graphql/product/resolvers.py [hunk 3]
```diff
@@ -73,9 +69,8 @@ def resolve_product_by_slug(info, product_slug, channel_slug):
 
 
 def resolve_products(
-    info, stock_availability=None, channel_slug=None, **_kwargs
+    info, user, stock_availability=None, channel_slug=None, **_kwargs
```
Comment (fowczarek) [inline saleor/graphql/product/resolvers.py:72]: Maybe?
```suggestion
    info, requestor, stock_availability=None, channel_slug=None, **_kwargs
```

File: saleor/graphql/product/resolvers.py [hunk 4]
```diff
@@ -99,10 +93,11 @@ def resolve_product_types(info, **_kwargs):
     return models.ProductType.objects.all()
 
 
-def resolve_product_variant_by_sku(info, sku, channel_slug):
-    requestor = get_user_or_app_from_context(info.context)
+def resolve_product_variant_by_sku(
+    info, sku, channel_slug, requestor, user_has_access_to_all
```
Comment (fowczarek) [inline saleor/graphql/product/resolvers.py:97]:
```suggestion
    info, sku, channel_slug, requestor, requestor_has_access_to_all
```

File: saleor/graphql/product/resolvers.py [hunk 5]
```diff
@@ -114,10 +109,11 @@ def resolve_product_variant_by_sku(info, sku, channel_slug):
     )
 
 
-def resolve_product_variants(info, ids=None, channel_slug=None) -> ChannelQsContext:
-    requestor = get_user_or_app_from_context(info.context)
+def resolve_product_variants(
+    info, access_to_all, requestor, ids=None, channel_slug=None
```
Comment (fowczarek) [inline saleor/graphql/product/resolvers.py:113]:
```suggestion
    info, requestor_has_access_to_all, requestor, ids=None, channel_slug=None
```

File: saleor/graphql/product/resolvers.py [hunk 6]
```diff
@@ -134,9 +130,9 @@ def resolve_report_product_sales(period, channel_slug=None) -> ChannelQsContext:
 
     # exclude draft and canceled orders
     exclude_status = [OrderStatus.DRAFT, OrderStatus.CANCELED]
-    qs = qs.exclude(order_lines__order__status__in=exclude_status).filter(
-        order_lines__order__channel__slug=str(channel_slug)
-    )
+    qs = qs.exclude(order_lines__order__status__in=exclude_status)
+    if channel_slug:
+        qs = qs.filter(order_lines__order__channel__slug=str(channel_slug))
```
Comment (fowczarek) [inline saleor/graphql/product/resolvers.py:135]: This resolver has no option to work without `channel_slug`. 
All order_lines in this `qs` should be in the same currency.

I think that `channel_slug` should be required for `reportProductSales` query.

File: saleor/graphql/product/schema.py [hunk 7]
```diff
@@ -302,14 +304,19 @@ def resolve_category(self, info, id=None, slug=None, **kwargs):
 
     def resolve_collection(self, info, id=None, slug=None, channel=None, **_kwargs):
         validate_one_of_args_is_in_query("id", id, "slug", slug)
-        if channel is None:
+        requestor = get_user_or_app_from_context(info.context)
+
+        user_has_access_to_all = models.Collection.objects.user_has_access_to_all(
```
Comment (fowczarek) [inline saleor/graphql/product/schema.py:309]:
```suggestion
        requestor_has_access_to_all = models.Collection.objects.user_has_access_to_all(
```

File: saleor/graphql/product/schema.py [hunk 8]
```diff
@@ -318,7 +325,9 @@ def resolve_collection(self, info, id=None, slug=None, channel=None, **_kwargs):
         )
 
     def resolve_collections(self, info, channel=None, *_args, **_kwargs):
-        if channel is None:
+        user = get_user_or_app_from_context(info.context)
+        user_has_access_to_all = models.Collection.objects.user_has_access_to_all(user)
```
Comment (fowczarek) [inline saleor/graphql/product/schema.py:329]:
```suggestion
        requestor_has_access_to_all = models.Collection.objects.user_has_access_to_all(requestor)
```

File: saleor/graphql/product/schema.py [hunk 9]
```diff
@@ -318,7 +325,9 @@ def resolve_collection(self, info, id=None, slug=None, channel=None, **_kwargs):
         )
 
     def resolve_collections(self, info, channel=None, *_args, **_kwargs):
-        if channel is None:
+        user = get_user_or_app_from_context(info.context)
```
Comment (fowczarek) [inline saleor/graphql/product/schema.py:328]:
```suggestion
        requestor = get_user_or_app_from_context(info.context)
```

File: saleor/graphql/product/schema.py [hunk 10]
```diff
@@ -332,21 +341,30 @@ def resolve_digital_contents(self, info, **_kwargs):
 
     def resolve_product(self, info, id=None, slug=None, channel=None, **_kwargs):
         validate_one_of_args_is_in_query("id", id, "slug", slug)
-        if channel is None:
+        requestor = get_user_or_app_from_context(info.context)
+        user_has_access_to_all = models.Collection.objects.user_has_access_to_all(
```
Comment (fowczarek) [inline saleor/graphql/product/schema.py:345]:
```suggestion
        requestor_has_access_to_all = models.Collection.objects.user_has_access_to_all(
```

File: saleor/graphql/product/schema.py [hunk 11]
```diff
@@ -332,21 +341,30 @@ def resolve_digital_contents(self, info, **_kwargs):
...
             product = resolve_product_by_slug(
-                info, product_slug=slug, channel_slug=channel
+                info, product_slug=slug, channel_slug=channel, requestor=requestor
             )
         return ChannelContext(node=product, channel_slug=channel) if product else None
 
     def resolve_products(self, info, channel=None, **kwargs):
-        if channel is None:
+        user = get_user_or_app_from_context(info.context)
+        user_has_access_to_all = models.Product.objects.user_has_access_to_all(user)
```
Comment (fowczarek) [inline saleor/graphql/product/schema.py:364]:
```suggestion
        requestor = get_user_or_app_from_context(info.context)
        requestor_has_access_to_all = models.Product.objects.user_has_access_to_all(requestor)
```

File: saleor/graphql/product/schema.py [hunk 12]
```diff
@@ -358,26 +376,42 @@ def resolve_product_variant(
         self, info, id=None, sku=None, channel=None,
     ):
         validate_one_of_args_is_in_query("id", id, "sku", sku)
-        if channel is None:
+        user = get_user_or_app_from_context(info.context)
+        user_has_access_to_all = models.Product.objects.user_has_access_to_all(user)
```
Comment (fowczarek) [inline saleor/graphql/product/schema.py:380]:
```suggestion
        requestor = get_user_or_app_from_context(info.context)
        requestor_has_access_to_all = models.Product.objects.user_has_access_to_all(requestor)
```

File: saleor/graphql/product/schema.py [hunk 13]
```diff
@@ -358,26 +376,42 @@ def resolve_product_variant(
...
+                channel_slug=channel,
+                requestor=user,
+                user_has_access_to_all=user_has_access_to_all,
             )
         return ChannelContext(node=variant, channel_slug=channel) if variant else None
 
     def resolve_product_variants(self, info, ids=None, channel=None, **_kwargs):
-        if channel is None:
+        requestor = get_user_or_app_from_context(info.context)
+        user_has_access_to_all = models.Product.objects.user_has_access_to_all(
```
Comment (fowczarek) [inline saleor/graphql/product/schema.py:400]:
```suggestion
        requestor_has_access_to_all = models.Product.objects.user_has_access_to_all(
```

File: saleor/graphql/product/types/products.py [hunk 14]
```diff
@@ -1011,16 +1011,21 @@ def resolve_url(root: models.Category, _info):
     @staticmethod
     def resolve_products(root: models.Category, info, channel=None, **_kwargs):
         requestor = get_user_or_app_from_context(info.context)
+        user_has_access_to_all = models.Product.objects.user_has_access_to_all(
```
Comment (fowczarek) [inline saleor/graphql/product/types/products.py:1014]:
```suggestion
        requestor_has_access_to_all = models.Product.objects.user_has_access_to_all(
```

**Review by fowczarek — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

**Pull request merged**